internal/pkg/logging/service.go

```diff
@@ -65,14 +67,29 @@ func (o WriteLogEventsOpts) limit() *int64 {
 	if o.Limit != nil {
 		return o.Limit
 	}
-	if o.StartTime != nil || o.EndTime != nil {
+	if o.hasTimeFilters() {
 		// If time filtering is set, then set limit to be maximum number.
 		// https://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_GetLogEvents.html#CWL-GetLogEvents-request-limit
 		return nil
 	}
 	return aws.Int64(defaultServiceLogsLimit)
 }
 
+func (o WriteLogEventsOpts) startTime(now func() time.Time) *int64 {
+	if o.StartTime != nil {
+		return o.StartTime
+	}
+	if o.Follow {
+		// Start following log events from current timestamp.
+		return aws.Int64(now().Unix() * 1000) // Multiplied by 1000 to convert the Unix() timestamp in seconds to milliseconds
```

Comment:
nit: [or alternatively](https://pkg.go.dev/time#Time.UnixMilli)...
```suggestion
		return aws.Int64(now().UnixMilli())
```

Reply:
wow I had no idea this method existed 😂 fixed!

I also changed it in other places in the codebase where we multiplied by 1000